packages/sdk/client-services/src/packlets/spaces/data-space-manager.ts

```diff
@@ -211,6 +200,53 @@ export class DataSpaceManager {
     return space;
   }
 
+  async isDefaultSpace(space: DataSpace): Promise<boolean> {
+    const rootDoc = await this._getSpaceRootDocument(space);
+    for (const object of Object.values(rootDoc.docSync()?.objects ?? {}) as ObjectStructure[]) {
+      if (object.system?.type?.itemId !== PropertiesType.typename) {
+        continue;
+      }
+      if (object.data?.[defaultKey] === this._signingContext.identityKey.toHex()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  async createDefaultSpace() {
+    const space = await this.createSpace();
+    const document = await this._getSpaceRootDocument(space);
+
+    // TODO(dmaretskyi): Better API for low-level data access.
+    const properties: ObjectStructure = {
+      system: {
+        type: encodeReference(getTypeReference(PropertiesType)!),
+      },
+      data: {
+        [defaultKey]: this._signingContext.identityKey.toHex(),
```

Comment:
is this for backwards compatibility? i suppose we woudn't need this with the new credential

Reply:
backward compatibility and was also thinking about using this for recovering from a case when a default space was created, but credential writing failed. we won't create another space but just write the missing credential on the next open. this however doesn't help if space was created but properties mutation failed, so not strongly against getting rid of it